src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] elements = (T[]) new Object[10];
+
+    private void ensureCapacity(int minCapacity) {
+        int currentCapacity = elements.length;
+        if (minCapacity > currentCapacity) {
+            int newCapacity = currentCapacity + (currentCapacity >> 1);
+            if (newCapacity < minCapacity) {
+                newCapacity = minCapacity;
+            }
+            elements = Arrays.copyOf(elements, newCapacity);
+        }
```

Comment:
it's better to avoid unnecessary use of bitwise operations. use constants instead
```suggestion
        int newCapacity = (int) (elementData.length * RESIZE_FACTOR);
        if (size == elementData.length) {
        ... do resize
        }
```